### ISSUE TITLE
{ error: 'ElasticsearchIllegalArgumentException[Failed to decode scrollId]; nested: IOException[Bad Base64 input charact er decimal 123 in array position 0]; ',   status: 400 }

### ISSUE THREAD

**Issue by DimonHo** [elastic/elasticsearch-js#549]
Sorry, my English is not good. here's my code:
```
esdao.searchScollFor130 = function() {
    esClient.search({
        index: esconf.index130,
        type: esconf.ttype,
        scroll: '30s',
        search_type: 'scan',
        body: {query: {match_all: {}}}
    }, function getMoreUntilDone(err, ress) {
        console.log(ress);
        if (ress.hits.total !== allresult.length) {
            esClient.scroll({
                scrollId: ress._scroll_id,
                scroll: '30s'
            }, getMoreUntilDone);
        } else {
            console.log('every "test" title', allTitles);
        }
    });
    console.log(allresult);
}
```
but 
![](https://segmentfault.com/img/bVN7CG?w=961&h=428)

Please answer me why this is？

**Comment by spalger** [comment]
Can you make sure that you have the latest version?

```sh
npm install --save elasticsearch@latest
```

**Comment by DimonHo** [comment]
yes,I have already installed it。
I think it may not be the JS problem, because the same mistake also appears in th "_plugin/head".
![2017-05-24_085105](https://cloud.githubusercontent.com/assets/3371641/26383102/7f51a296-4063-11e7-8439-1b61db81932a.png)
![2017-05-24_085304](https://cloud.githubusercontent.com/assets/3371641/26383106/827e9410-4063-11e7-9b0c-b9e2c436a13b.png)

but，I know the cause of the problem。
Change scroll_id, Remove “”。
scroll_id="cXVlcnlUaGVuRmV0Y2g7MjsxNjY1MjppZExGcjRFX1RrQzNBTnVybC10V2N3OzE2NjUzOmlkTEZyNEVfVGtDM0FOdXJsLXRXY3c7MDs="  
==>
scroll_id=cXVlcnlUaGVuRmV0Y2g7MjsxNjY1MjppZExGcjRFX1RrQzNBTnVybC10V2N3OzE2NjUzOmlkTEZyNEVfVGtDM0FOdXJsLXRXY3c7MDs=

---

but in nodejs ,Double quotes will be added automatically。
```JavaScript
var allresult= [];
esdao.searchScollFor130 = function() {
	esClient.search({
		index: esconf.index130,
		type: esconf.ttype,
		scroll: '30s',
		search_type: 'scan',
		body: {query: {match_all: {}}}
	}, function getMoreUntilDone(error, response) {
		console.log("================================");
		console.log(response);
		console.log(error);
		console.log("================================");
		response.hits.hits.forEach(function(hit) {
			var jsonobject = {};
			jsonobject.docTitle = hit.fields.docTitle;
			jsonobject.issn = hit.fields.issn;
			allresult.push(jsonobject);
		});
		if (response.hits.total !== allresult.length) {
			esClient.scroll({
				scrollId: response._scroll_id,
				scroll: '30s'
			}, getMoreUntilDone);
		} else {
			console.log('every "test" title', allresult);
		}
	});
	console.log(allresult);
}
```
![image](https://cloud.githubusercontent.com/assets/3371641/26383998/df05c686-4068-11e7-9705-df5183384707.png)

**Comment by DimonHo** [comment]
@spalger haha~
I have solved the problem。 Inspired by #546。
Because my server version is 1.2.1，When creating a client，add  apiVersion: "0.90"
```
var esClient = new elasticsearch.Client({
  host: esconf.host130,
  apiVersion: "0.90",
  log: 'error'
  });
```

**Comment by spalger** [comment]
Glad you figured that out :)
